Core/Core/CourseSync/CourseSyncDownloader/Model/CourseSyncPeopleInteractor.swift

```diff
@@ -0,0 +1,148 @@
+                .getEntities()
+                .mapToVoid()
+                .eraseToAnyPublisher()
+        }
+
+    private static func fetchSingleUser(context: Context, userID: String) -> AnyPublisher<Void, Error> {
+        ReactiveStore(useCase: GetCourseContextUser(context: context, userID: userID))
+            .getEntities()
+            .mapToVoid()
+            .eraseToAnyPublisher()
+    }
+
+    private static func fetchUsers(context: Context) -> AnyPublisher<[User], Error> {
+        ReactiveStore(useCase: GetContextUsers(context: context))
+            .getEntities()
+            .eraseToAnyPublisher()
+    }
+
+    private static func fetchGroup(context: Context) -> AnyPublisher<Void, Error> {
+        ReactiveStore(useCase: GetGroup(groupID: context.id))
+            .getEntities()
+            .mapToVoid()
+            .eraseToAnyPublisher()
+    }
+
+    private static func fetchCurrentGradingPeriodId(courseID: String) -> AnyPublisher<String?, Error> {
+        ReactiveStore(useCase: GetGradingPeriods(courseID: courseID))
+            .getEntities()
+            .map {
+                $0.current?.id
+            }
+            .eraseToAnyPublisher()
+    }
+
+    private static func fetchSections(courseID: String) -> AnyPublisher<Void, Error> {
+        ReactiveStore(useCase: GetCourseSections(courseID: courseID))
+            .getEntities()
+            .mapToVoid()
+            .eraseToAnyPublisher()
+    }
+
+    private static func fetchEnrollments(context: Context, currentGradingPeriodID: String?, userID: String) -> AnyPublisher<Void, Error> {
+        Future { promise in
+            let env = AppEnvironment.shared
+            guard context.id == env.currentSession?.userID else { return promise(.failure(NSError.instructureError("User ID not found.")))}
+            let request = GetEnrollmentsRequest(context: context, gradingPeriodID: currentGradingPeriodID, states: [ .active ])
+            AppEnvironment.shared.api.exhaust(request) { (enrollments, _, error) in performUIUpdate {
+
+                if let error = error {
+                    promise(.failure(NSError.instructureError("Unexpected error from API.")))
+                    return
+                }
+
+                let apiEnrollment = enrollments?.first {
+                    $0.id != nil &&
+                    $0.enrollment_state == .active &&
+                    $0.user_id.value == userID
+                }
+                if let apiEnrollment = apiEnrollment, let id = apiEnrollment.id?.value {
+                    let databaseContext = env.database.viewContext
+                    let enrollment: Enrollment = databaseContext.first(where: #keyPath(Enrollment.id), equals: id) ?? databaseContext.insert()
+                    enrollment.update(fromApiModel: apiEnrollment, course: nil, in: databaseContext)
+                }
+                promise(.success(()))
+            }}
```

Comment:
Couldn't we create a use case for this logic?

Reply:
and we did.